exercise/static/exercise/user_results.js

```diff
@@ -0,0 +1,102 @@
+	if (autoScrollBehavior === "instant") {
+		$("#auto-scroll-behavior-instant").show();
+	} else if (autoScrollBehavior === "off") {
+		$("#auto-scroll-behavior-off").show();
+	} else {
+		$("#auto-scroll-behavior-smooth").show();
+		localStorage.setItem("autoScrollBehavior", "smooth"); // Scroll smoothly by default
+	}
+
+	// Register event handlers to all three scroll behavior buttons
+	$("#cycle-auto-scroll-behavior button").on("click", function(event) {
+		$(this).hide();
+		const autoScrollBehavior = localStorage.getItem("autoScrollBehavior");
+		if (autoScrollBehavior === "instant") {
+			$("#auto-scroll-behavior-off").show();
+			localStorage.setItem("autoScrollBehavior", "off");
+		} else if (autoScrollBehavior === "off") {
+			$("#auto-scroll-behavior-smooth").show();
+			localStorage.setItem("autoScrollBehavior", "smooth");
+		} else {
+			$("#auto-scroll-behavior-instant").show();
+			localStorage.setItem("autoScrollBehavior", "instant");
+		}
+	});
+
+	$('.filter-categories button').on("click", function(event) {
+		var button = $(this);
+		var id = button.attr("data-category");
+		if (button.hasClass("active")) {
+			button.removeClass("active").find("span").removeClass("glyphicon-unchecked").addClass("glyphicon-check");
+			$('.module-panel tr[data-category="' + id + '"]').removeClass("hide");
+		} else {
+			button.addClass("active").find("span").removeClass("glyphicon-check").addClass("glyphicon-unchecked");
+			$('.module-panel tr[data-category="' + id + '"]').addClass("hide");
+		}
+		$('.module-panel').each(function(index, panel) {
+			var mod = $(panel);
+			if (mod.find("tr:not(.hide)").length > 0) {
+				mod.removeClass("hide");
+			} else {
+				mod.addClass("hide");
+			}
+		});
+	});
+
+	// Get current course news from the document and previously shown course news from local storage
+	const news = [];
+	$(".news-panel > .list-group > .list-group-item > .list-group-item-text").each(function() {
+		news.push($(this)[0].innerText);
+	});
+	const course = "{{course|safe}}";
+	const allPreviousNews = JSON.parse(localStorage.getItem("courseNews")) || {};
+	const previousNews = course in allPreviousNews ? allPreviousNews[course] : [];
+	const currentNews = news.length > 0 ? news : previousNews;
+	if (news.length > 0) {
+		const allCurrentNews = { ...allPreviousNews, [course]: currentNews };
+		localStorage.setItem("courseNews", JSON.stringify(allCurrentNews));
+	}
+
+	function autoScroll() {
+		const autoScrollBehavior = localStorage.getItem("autoScrollBehavior");
+		if (autoScrollBehavior !== "off") {
+			// Scroll to the first open module if there are no new news items
+			const openModules = $(".module-panel:not(.module-to-collapse)");
+			if (openModules.length > 0 && JSON.stringify(currentNews) === JSON.stringify(previousNews)) {
+				const firstHeadingId = openModules[0].querySelector(".panel-heading").id;
+				document.getElementById(firstHeadingId).scrollIntoView({ behavior: autoScrollBehavior });
```

Comment:
Why does this query the element, take its id, and then query it again? Seems to me like `openModules[0].querySelector(".panel-heading").scrollIntoView(...` would work fine. This also makes the heading id unneccesary in the html.

Reply:
You're right! Fixed.